aws/resource_aws_eip_test.go

```diff
@@ -306,6 +306,35 @@ func TestAccAWSEIP_tags(t *testing.T) {
 	})
 }
 
+func TestAccAWSEIP_PublicIpv4Pool(t *testing.T) {
```

Comment:
Thanks so much for adding this acceptance test! It seems like we might want to either create one test for setting `"amazon"` and another for BYOIP that purposefully calls `t.Skip()` when the environment variable is missing/empty or just convert this test to call `t.Skip()` when the environment variable is missing/empty. There are a few other acceptance tests in the repository that have this type of setup, e.g. `TestAccAWSCognitoUserPoolDomain_custom`

Reply:
That's a great idea and clean solution which I have not realized when was doing this change. I will implement two separate tests for default and custom values of that attribute. But that would only work and valid for VPC EIPs.